src/Compilers/CSharp/Portable/Compiler/MethodCompiler.cs

```diff
@@ -948,7 +948,8 @@ public override object VisitField(FieldSymbol symbol, TypeCompilationState argum
                         return semanticModel;
                     });
 
-                    _compilation.EventQueue.Enqueue(new SymbolDeclaredCompilationEvent(_compilation, methodSymbol, lazySemanticModel));
+                    MethodSymbol symbolToProduce = methodSymbol.PartialDefinitionPart ?? methodSymbol;
+                    _compilation.EventQueue.Enqueue(new SymbolDeclaredCompilationEvent(_compilation, symbolToProduce, lazySemanticModel));
```

Comment:
I am not sure this code is doing the right thing now. The partial definition can be in a different syntax tree, but we are passing semantic model for implementation part.